x/evm/keeper/msg_server.go

```diff
@@ -131,13 +124,21 @@ func (server msgServer) writeReceipt(ctx sdk.Context, tx *ethtypes.Transaction,
 		TxType:            uint32(tx.Type()),
 		CumulativeGasUsed: cumulativeGasUsed,
 		TxHashHex:         tx.Hash().Hex(),
-		ContractAddress:   contractAddr.Hex(),
 		GasUsed:           usedGas,
 		BlockNumber:       uint64(ctx.BlockHeight()),
 		TransactionIndex:  uint32(ctx.TxIndex()),
 		EffectiveGasPrice: tx.GasPrice().Uint64(),
 	}
 
+	if msg.To == nil {
+		receipt.ContractAddress = crypto.CreateAddress(msg.From, msg.Nonce).Hex()
```

Comment:
is it expected that a nil "To" has the side effect of creating an address?

Reply:
yeah it's a `contract deployment transaction`: https://ethereum.org/en/developers/docs/transactions/#types-of-transactions